### DISCUSSION TITLE
Modeling different types of loading

### DISCUSSION THREAD

**Discussion by bahrami4444** [lululxvi/deepxde#1786] [General]
Hello

I want to model a 2D problem with DeepXDE but I couldn't find any example similar to my problem. Is this geometry and this type of loading possible to be modeled whit DeepXDE? Does it require backend editing? Does anyone have similar code?

In the first case, it is a 2D problem where the load is applied only to parts of the top of the model (not all parts of the upper boundary).

![case 1](https://github.com/lululxvi/deepxde/assets/81992995/1aff0457-4814-4f8a-8c93-24d35aa7d89d)

The second case is the same as the first one, but it is a dynamic load (Time-dependent).

![case 2](https://github.com/lululxvi/deepxde/assets/81992995/2877e89b-7af2-4f89-b2a1-f9fb66855fb2)

The third case is both a dynamic load and moving load. That is, loading is a function of t and x (time and position).

![case 3](https://github.com/lululxvi/deepxde/assets/81992995/f0617d62-2e40-44b6-8fa4-ad79a6618546)

I would be grateful if someone could guide me about the possibility of modeling each of the three cases mentioned. Can all three cases be modeled with DeepXDE? Does anyone have sample code?

**Comment by lululxvi** [comment]
See
- https://github.com/lu-group/pinn-heterogeneous-material 
- https://github.com/lu-group/pinn-material-identification

**Reply by bahrami4444** [reply]
Thanks @lululxvi 🙏🙏
